### PR TITLE
[PhpUnitBridge] Fix compatibility to PHPUnit 9

### LINKED ISSUE #35662: [PhpUnitBridge] Add compatibility to PHPUnit 9
**Symfony version(s) affected**: 4.4.4 (Tested version)

**Description**  
Impossible to use PHPUnit 9 with the actual version of Symfony/PHPUnit-bridge
Because of: https://github.com/sebastianbergmann/phpunit/pull/3914

**How to reproduce**  
In a Symfony project (Version 4.4.4)
Change `SYMFONY_PHPUNIT_VERSION` to 9 (in phpunit.xml or env var)
Launch your test you will get: `PHP Fatal error:  Uncaught Error: Call to undefined method PHPUnit\TextUI\Configuration\Configuration::getInstance()`

**Possible Solution**  
I suggest to : 

- add a new `CommandForV9` file with these changes : 

```diff
<?php

/*
 * This file is part of the Symfony package.
 *
 * (c) Fabien Potencier <fabien@symfony.com>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

namespace Symfony\Bridge\PhpUnit\Legacy;

use PHPUnit\TextUI\Command as BaseCommand;
- use PHPUnit\Util\Configuration;
+ use PHPUnit\TextUI\Configuration\Configuration;
+ use PHPUnit\TextUI\Configuration\Registry;
use PHPUnit\TextUI\TestRunner as BaseRunner;
use Symfony\Bridge\PhpUnit\SymfonyTestsListener;

/**
 * {@inheritdoc}
 *
 * @internal
 */
- class CommandForV6 extends BaseCommand
+ class CommandForV9 extends BaseCommand
{
    /**
     * {@inheritdoc}
     */
    protected function createRunner(): BaseRunner
    {
        $this->arguments['listeners'] = isset($this->arguments['listeners']) ? $this->arguments['listeners'] : [];

        $registeredLocally = false;

        foreach ($this->arguments['listeners'] as $registeredListener) {
            if ($registeredListener instanceof SymfonyTestsListener) {
                $registeredListener->globalListenerDisabled();
                $registeredLocally = true;

                break;
            }
        }

        if (isset($this->arguments['configuration'])) {
            $configuration = $this->arguments['configuration'];
            if (!$configuration instanceof Configuration) {
-                $configuration = Configuration::getInstance($this->arguments['configuration']);
+                $configuration = Registry::getInstance()->get($this->arguments['configuration']);
            }
-            foreach ($configuration->getListenerConfiguration() as $registeredListener) {
+            foreach ($configuration->listeners() as $registeredListener) {
-                 if ('Symfony\Bridge\PhpUnit\SymfonyTestsListener' === ltrim($registeredListener['class'], '\\')) {
+                 if ('Symfony\Bridge\PhpUnit\SymfonyTestsListener' === ltrim($registeredListener->className(), '\\')) {
                    $registeredLocally = true;

                    break;
                }
            }
        }

        if (!$registeredLocally) {
            $this->arguments['listeners'][] = new SymfonyTestsListener();
        }

        return parent::createRunner();
    }
}
```
And add the new file in https://github.com/symfony/phpunit-bridge/blob/master/TextUI/Command.php to switch to the right Command depending on the version.

At this moment I don't know if more changes are required to be fully compatible with PHPUnit 9 but in my case, it's enough to be able to test my code.

### PR DESCRIPTION
**Pull request by Hakihiro**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no
| Tickets       | Fix #35662
| License       | MIT
| Doc PR        |  -

Adding the possibility to use PHPUnit 9.

### CODE REVIEW

File: src/Symfony/Bridge/PhpUnit/TextUI/Command.php [hunk 1]
```diff
@@ -13,8 +13,10 @@
 
 if (version_compare(\PHPUnit\Runner\Version::id(), '6.0.0', '<')) {
     class_alias('Symfony\Bridge\PhpUnit\Legacy\CommandForV5', 'Symfony\Bridge\PhpUnit\TextUI\Command');
-} else {
+} elseif (version_compare(\PHPUnit\Runner\Version::id(), '6.0.0', '>=') && version_compare(\PHPUnit\Runner\Version::id(), '9.0.0', '<')) {
```
Comment (stof) [inline src/Symfony/Bridge/PhpUnit/TextUI/Command.php]: the first part of the condition is useless. The `elseif` already ensures it is always true.

File: src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php [hunk 2]
```diff
@@ -93,7 +93,10 @@
     }
 };
 
-if (PHP_VERSION_ID >= 70200) {
+if (PHP_VERSION_ID >= 70300) {
```
Comment (stof) [inline src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php]: this is a BC break (people using the bridge will be upgraded to the PHPUnit 9 version, with all its BC breaks impacting the testsuite) in the LTS version of Symfony. that's not good. this PR must go in master.

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php (hunk 2 above)
Comment (Hakihiro) [reply src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php]: Is it better to remove this condition to let everyone have the option? Or maybe to push it to master in a sperate PR?

I think the first part of this PR is enough for the compatibility case PHPUnit9/SF4.4

Comment (nicolas-grekas) [reply src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php]: Splitting might be a good idea, the other changes in the PR are OK for 4.4
But looking at the changelog, I'm not sure we need to bother ppl with a bump of the default version honeslty.

Comment (Hakihiro) [reply src/Symfony/Bridge/PhpUnit/bin/simple-phpunit.php]: PR is updated. 

I removed the bump of the default version, to make this PR compatible with 4.4 branch.

Let me know if you want a PR to master that enable the default version.

Thanks, both of you for your feedback.

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @Hakihiro.

**Pull request merged**